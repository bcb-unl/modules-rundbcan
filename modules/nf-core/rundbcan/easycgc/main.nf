process RUNDBCAN_EASYCGC {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/dbcan:5.0.4--pyhdfd78af_0' :
        'biocontainers/dbcan:5.0.4--pyhdfd78af_0' }"
    input:
    tuple val(meta), path(input_raw_data)
    tuple val(meta2), path(input_gff), val (gff_type)
    path dbcan_db

    output:
    tuple val(meta), path("${prefix}/overview.tsv"), emit: cazyme_annotation
    tuple val(meta), path("${prefix}/dbCAN_hmm_results.tsv"), emit: dbcanhmm_results
    tuple val(meta), path("${prefix}/dbCANsub_hmm_results.tsv"), emit: dbcansub_results
    tuple val(meta), path("${prefix}/diamond.out"), emit: dbcandiamond_results
    tuple val(meta), path("${prefix}/cgc.gff"), emit: cgc_gff
    tuple val(meta), path("${prefix}/cgc_standard_out.tsv"), emit: cgc_standard_out
    tuple val(meta), path("${prefix}/diamond.out.tc"), emit: diamond_out_tc
    tuple val(meta), path("${prefix}/TF_hmm_results.tsv"), emit: tf_hmm_results
    tuple val(meta), path("${prefix}/STP_hmm_results.tsv"), emit: stp_hmm_results
    tuple val(meta), path("${prefix}/total_cgc_info.tsv"), emit: total_cgc_info

    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}_dbcan_cgc"
    def VERSION = '5.0.4'

    """

    run_dbcan easy_CGC \\
        --mode protein \\
        --db_dir ${dbcan_db} \\
        --input_raw_data ${input_raw_data} \\
        --output_dir ${prefix} \\
        --input_gff ${input_gff} \\
        --gff_type ${gff_type} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}_dbcan_cgc"
    def VERSION = '5.0.4'
    """
    mkdir -p ${prefix}
    touch ${prefix}/overview.tsv
    touch ${prefix}/dbCAN_hmm_results.tsv
    touch ${prefix}/dbCANsub_hmm_results.tsv
    touch ${prefix}/diamond.out
    touch ${prefix}/cgc.gff
    touch ${prefix}/cgc_standard_out.tsv
    touch ${prefix}/diamond.out.tc
    touch ${prefix}/TF_hmm_results.tsv
    touch ${prefix}/STP_hmm_results.tsv
    touch ${prefix}/total_cgc_info.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: $VERSION
    END_VERSIONS
    """
}
