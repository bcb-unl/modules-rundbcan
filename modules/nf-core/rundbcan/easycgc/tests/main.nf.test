nextflow_process {

    name "Test Process RUNDBCAN_EASYCGC"
    script "../main.nf"
    process "RUNDBCAN_EASYCGC"

    tag "modules"
    tag "modules_nfcore"
    tag "rundbcan"
    tag "rundbcan/database"
    tag "rundbcan/easycgc"


    test("easycgc - simplified") {

        setup {
            run("RUNDBCAN_DATABASE"){
                script "../../database/main.nf"
                process {
                    """
                    """
                }

            }
        }

        when {
            process {
                """
                input[0] = [
                    [id:'test_dbcan'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/proteome.fasta', checkIfExists: true)
                ]


                input[1] = [
                    [id:'test_dbcan'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/gff/test1.gff', checkIfExists: true)
                    ,"prodigal"
                    ]

                input[2] = RUNDBCAN_DATABASE.out.dbcan_db
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert path(process.out.cazyme_annotation.get(0).get(1)).exists() },
                { assert path(process.out.dbcanhmm_results.get(0).get(1)).exists() },
                { assert path(process.out.dbcansub_results.get(0).get(1)).exists()  },
                { assert path(process.out.dbcandiamond_results.get(0).get(1)).exists()},
                { assert path(process.out.cgc_gff.get(0).get(1)).exists() },
                { assert path(process.out.cgc_standard_out.get(0).get(1)).exists() },
                { assert path(process.out.diamond_out_tc.get(0).get(1)).exists() },
                { assert path(process.out.tf_hmm_results.get(0).get(1)).exists() },
                { assert path(process.out.stp_hmm_results.get(0).get(1)).exists() },
                { assert path(process.out.total_cgc_info.get(0).get(1)).exists() }
            )
        }
    }

    test("easycgc - stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [[id: 'stub'], file('stub')]
                input[1] = [[id: 'stub'], file('stub.gff'), "prodigal"]
                input[2] = file('stub_db')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match()},
                { assert path(process.out.cazyme_annotation.get(0).get(1)).exists() },
                { assert path(process.out.dbcanhmm_results.get(0).get(1)).exists() },
                { assert path(process.out.dbcansub_results.get(0).get(1)).exists()  },
                { assert path(process.out.dbcandiamond_results.get(0).get(1)).exists()  },
                { assert path(process.out.cgc_gff.get(0).get(1)).exists() },
                { assert path(process.out.cgc_standard_out.get(0).get(1)).exists() },
                { assert path(process.out.diamond_out_tc.get(0).get(1)).exists() },
                { assert path(process.out.tf_hmm_results.get(0).get(1)).exists() },
                { assert path(process.out.stp_hmm_results.get(0).get(1)).exists() },
                { assert path(process.out.total_cgc_info.get(0).get(1)).exists() }
            )

        }
    }
}

