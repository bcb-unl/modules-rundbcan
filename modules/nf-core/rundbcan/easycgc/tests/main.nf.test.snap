{
    "easycgc - simplified": {
        "content": [
            [
                "versions.yml:md5,0f16967f8e1d900999a59c516abc360e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-21T14:18:33.375969"
    },
    "easycgc - stub": {
        "content": [
            [
                "versions.yml:md5,0f16967f8e1d900999a59c516abc360e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-21T14:18:37.031568"
    }
}